src/core/multifeaturelistmodelbase.cpp

```diff
@@ -328,6 +328,15 @@ bool MultiFeatureListModelBase::canDuplicateSelection()
   return !vlayer->readOnly() && ( vlayer->dataProvider()->capabilities() & QgsVectorDataProvider::AddFeatures );
 }
 
+bool MultiFeatureListModelBase::canMoveSelection()
+{
+  if ( mSelectedFeatures.isEmpty() )
+    return false;
+
+  QgsVectorLayer *vlayer = mSelectedFeatures[0].first;
```

Comment:
Is it enough to check the first feature only?

Reply:
I've fixed the 'out of sync layer filter' issue you stumbled upon which led you to collide worlds and select features from multiple layers.